lib/driver.js

```diff
@@ -450,16 +451,14 @@ class YouiEngineDriver extends BaseDriver {
       retryCount++;
       connected = await connectedPromise;
 
-      if (!connected && errno === 'ECONNREFUSED') {
+      if (!connected && (errno === 'ECONNREFUSED' || errno === -61)) {
         logger.debug('Connection refused, sleeping...');
         await sleep(2000);
         errno = 'EOK';
       }
 
-      if (!connected && retryCount === MAX_RETRY_COUNT - 1) {
-        logger.errorAndThrow(
-          'Failed to connect ' + MAX_RETRY_COUNT + ' times. Aborting.'
-        );
+      if (!connected && retryCount === this.maxRetryCount) {
```

Comment:
This happens after `retryCount++` so it was actually retrying one less time than expected, if you set maxRetryCount at 3 it'd only do 2 attempts

Reply:
Ironically I was just looking at this logic today as well. I'll keep it in your PR.